clouddriver-alicloud/src/main/java/com/netflix/spinnaker/clouddriver/alicloud/deploy/ops/DeleteAliCloudLoadBalancerClassicAtomicOperation.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019 Alibaba Group.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.alicloud.deploy.ops;
+
+import com.aliyuncs.IAcsClient;
+import com.aliyuncs.exceptions.ClientException;
+import com.aliyuncs.exceptions.ServerException;
+import com.aliyuncs.slb.model.v20140515.DeleteLoadBalancerRequest;
+import com.aliyuncs.slb.model.v20140515.DeleteLoadBalancerResponse;
+import com.aliyuncs.slb.model.v20140515.DescribeLoadBalancersRequest;
+import com.aliyuncs.slb.model.v20140515.DescribeLoadBalancersResponse;
+import com.netflix.spinnaker.clouddriver.alicloud.common.ClientFactory;
+import com.netflix.spinnaker.clouddriver.alicloud.deploy.description.UpsertAliCloudLoadBalancerDescription;
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation;
+import java.util.List;
+
+public class DeleteAliCloudLoadBalancerClassicAtomicOperation implements AtomicOperation<Void> {
+
+  private final UpsertAliCloudLoadBalancerDescription description;
+
+  private final ClientFactory clientFactory;
+
+  public DeleteAliCloudLoadBalancerClassicAtomicOperation(
+      UpsertAliCloudLoadBalancerDescription description, ClientFactory clientFactory) {
+    this.description = description;
+    this.clientFactory = clientFactory;
+  }
+
+  @Override
+  public Void operate(List priorOutputs) {
+
+    IAcsClient client =
+        clientFactory.createClient(
+            description.getRegion(),
+            description.getCredentials().getAccessKeyId(),
+            description.getCredentials().getAccessSecretKey());
+    DescribeLoadBalancersResponse.LoadBalancer loadBalancerT = null;
+    DescribeLoadBalancersRequest queryRequest = new DescribeLoadBalancersRequest();
+    queryRequest.setLoadBalancerName(description.getLoadBalancerName());
+    DescribeLoadBalancersResponse queryResponse;
+    try {
+      queryResponse = client.getAcsResponse(queryRequest);
+      description.setLoadBalancerId(queryRequest.getLoadBalancerId());
+
+      if (queryResponse.getLoadBalancers().size() > 0) {
+        loadBalancerT = queryResponse.getLoadBalancers().get(0);
+      }
+
+    } catch (ServerException e) {
+      e.printStackTrace();
+      throw new IllegalStateException(e.getMessage());
```

Comment:
Create specific exception classes for the Alicloud provider: https://github.com/spinnaker/kork/commit/b559eda1d0fa8131b5e3f769cdcba1f523cc97e1

Reply:
Currently, I think I can not do it because of its parent interface does not expose it. Is there any other ideas?

Reply:
```
public class AlibabaException extends IntegrationException {}

// ...

throw new AlibabaException(e.getMessage());
```

Reply:
This would require depending on `kork-exceptions`, which I presume is already part of your classpath, but if not, you should add it.